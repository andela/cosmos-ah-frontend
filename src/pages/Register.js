import React from 'react';
import { Form, Button } from 'semantic-ui-react';
import styled, { css } from 'styled-components';
// import img from 'https://res.cloudinary.com/drmrayjr0/image/upload/v1556868879/bg2_sguhi2.png';
import Icon from '../components/Icon';


const ButtonWrapper = styled.button`
  border-radius: 3px;
  border: none;
  color: #3A8FDD;
  margin: 0 1em;
  padding: 0.25em 1em;

  ${props => props.primary
    && css`
      background: #3A8FDD;
      color: white;
    `};
`;

const Container = styled.div`
  margin: 0 auto;
  width: 70%;
  height: 70%;
  display: grid;
  grid-template-columns: 35% 65%;
  grid-gap: 30px;
  border: 2px solid #3A8FDD;
  content-align: center;
  background-image: url('https://res.cloudinary.com/drmrayjr0/image/upload/v1556869811/bg2_ukfjbe.png');
`;

const H1 = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 100%;
  text-align: center;
`;

const Center = styled.div`
  margin: 0 auto;
  content-align: center;
`;

const ContainerCenter = styled.div`
  margin: 0 auto;
  text-align: center;
  width: 100%;
`;

const Background = styled.div`
background-image: url('https://res.cloudinary.com/drmrayjr0/image/upload/v1556869811/bg1_fskesq.png');
`;

const Input = styled.input``;
const Paragraph = styled.p``;

const Register = () => (
  <Container>
    <Background>
      <div>
        <div>LOGO</div>
        <H1>
          <div>
            <h2>"Character is like a tree and reputation is like shadow.
         The shadow is what we think of it; the tree is the real thing."</h2>
            <h3>Abraham Lincoln</h3>
          </div>
        </H1>
      </div>
    </Background>
    <Center>
      <div>
        <ContainerCenter>
          <h1>Create Your Account</h1>
          <Icon />
          <p>or register using your email address</p>
        </ContainerCenter>
        <Form>
          <Form.Input label='Full Name' type='text' placeholder='Full Name' width={16} />
          <Form.Input label='Email Address' type='email' placeholder='Email Address' width={16} />
          <Form.Input label='Password' type='password' placeholder='Password' width={16} />
          <Form.Input label='Confirm Password' type='password' placeholder='Confirm Password' width={16} />
          <Form.Checkbox label='Enable Email Notifications About Product And Services.' width={16} />
          <div style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
          }}>
            <Button type='submit' content="Sign Up" primary />
            <span style={{
              margin: '0 20px',
              fontWeight: 'bold'
            }}>or</span>
            <Button type='submit' content="Sign In" basic />
          </div>
        </Form>
      </div>
      <div style={{
        textAlign: 'center',
      }}>
        <p>By signing up, you agree to Authors Haven</p>
        <p>Terms and Conditions and Privacy Policy</p>
        <p>&copy;2019 All Right Reserved.</p>
      </div>
    </Center>
  </Container>
);

export default Register;
