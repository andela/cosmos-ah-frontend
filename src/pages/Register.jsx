import React from 'react';
import { Form, Button } from 'semantic-ui-react';
import styled from 'styled-components';
import SocialMediaButton from '../components/SocialMediaButton';


const Center = styled.div`
  text-align: center;
`;

const TermsStyle = styled(Center)`
  color: #9B9B9B;
  padding-bottom: 20px;
`;

const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 2fr;
  background: #FFFFFF;
  height: 100vh;
  
  @media screen and (max-width: 768px) {
    display: block;
  }
`;

const MyButton = styled(Button)`
  &&& {
    background: ${props => (props.primary ? '#3A8FDD' : '#FFFFFF')};
    color: ${props => (props.primary ? '#FFFFFF' : 'grey')};
    width: ${props => (props.primary ? '40%' : '')};
    border-radius: 5px;
    border: none;
    margin: 10px;
    padding: 8px 0px;
    cursor: pointer;
    font-size: 16px;
  }
`;

const Aside = styled(Center)`
  margin-bottom: auto;
  color: #FFFFFF;
  padding: 35px;
`;

const ContainerCenter = styled(Center)`
  color: #3A8FDD;
  font-size: 18px;
  margin-top: 100px;
  @media screen and (max-width: 768px) {
    margin-top: 50px;
  }
`;

const Background = styled.div`
  background-image: url('https://res.cloudinary.com/drmrayjr0/image/upload/v1556869811/bg1_fskesq.png');
  display: flex;
  flex-direction: column;
  @media screen and (max-width: 768px) {
    display: none;
  }
`;

const FormInput = styled(Form)`
  &&& {
    width: 30%;
  }
  input {
    background-color: #E8E8E8 !important;
    border-radius: 10px !important;
    border: none !important;
  }
  label {
    color: #3A8FDD !important;
    }
  }
  @media screen and (max-width: 768px) {
    &&& {
      width: 90%;
    }
  }
  @media (min-width: 769px) and (max-width: 907px) {
    &&& {
      width: 50%;
    }
  }
`;
const FormContainer = styled.div`
  padding-top: 15px;
  display: flex;
  justify-content: center;
`;

const Header = styled.div`
  margin-top: 50px;
  margin-bottom: auto;
  h1 {
    font-size: 40px;
  }
  img {
    margin-left: 35px;
  }
  @media screen and (max-width: 768px) {
    h1 {
      font-size: 30px;
    }
  }
`;

const Register = () => (
  <Container>
    <Background>
      <Header>
        <div>
          <a href='/'><img src='https://res.cloudinary.com/drmrayjr0/image/upload/v1556992917/Screenshot_2019-04-18_at_1.45.17_PM_aurl4d.png' width={150} /></a>
        </div>
      </Header>
      <Aside>
        <h2>"Character is like a tree and reputation is like shadow.
        The shadow is what we think of it; the tree is the real thing."</h2>
        <h3>Abraham Lincoln</h3>
      </Aside>
    </Background>
    <div>
      <ContainerCenter>
        <Header>
          <h1>Create Your Account</h1>
        </Header>
        <SocialMediaButton />
        <p>or register using your email address</p>
      </ContainerCenter>
      <FormContainer>
        <FormInput data-testid='registerForm'>
            <Form.Input icon={{ name: 'user', color: 'blue' }} iconPosition='left' placeholder='Full Name' />
            <Form.Input icon={{ name: 'user', color: 'blue' }} iconPosition='left' placeholder='Username' />
            <Form.Input icon={{ name: 'mail', color: 'blue' }} iconPosition='left' placeholder='Email Address' type='email' />
            <Form.Input icon={{ name: 'lock', color: 'blue' }} iconPosition='left' placeholder='Password' type='password' />
            <Form.Input icon={{ name: 'lock', color: 'blue' }} iconPosition='left' placeholder='Confirm Password' type='password' />
            <Form.Checkbox label='Enable Email Notifications About Product And Services.' />
            <Center>
              <MyButton primary>Sign Up</MyButton> Or <MyButton>Sign In</MyButton>
            </Center>
        </FormInput>
      </FormContainer>
      <TermsStyle>
        <div>
          <p>By signing up, you agree to Authors Haven
            <br />
            Terms and Conditions and Privacy Policy
          <br />
            &copy;2019 All Right Reserved.</p>
        </div>
      </TermsStyle>
    </div>
  </Container>
);

export default Register;
