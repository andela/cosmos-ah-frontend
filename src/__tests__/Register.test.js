import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Register from '../pages/Register';

afterEach(cleanup);

test('<Register /> component', () => {
  const registrationPage = render(<Register />);
  expect(registrationPage).toBeTruthy();
  expect(registrationPage.getByText('Sign Up').tagName).toBe('BUTTON');
  expect(registrationPage.getByText('Sign In').tagName).toBe('A');
});

test('renders the Form', () => {
  const { getByPlaceholderText } = render(<Register />);
  expect(getByPlaceholderText('Full Name')).toBeTruthy();
  expect(getByPlaceholderText('Username')).toBeTruthy();
  expect(getByPlaceholderText('Email Address')).toBeTruthy();
  expect(getByPlaceholderText('Password')).toBeTruthy();
  expect(getByPlaceholderText('Confirm Password')).toBeTruthy();
});

test('renders the Header', () => {
  const registrationPage = render(<Register />);
  const Header = registrationPage.getByText('Create Your Account');
  expect(Header.textContent).toBe('Create Your Account');
});

test('renders the Component', () => {
  const registrationPage = render(<Register />);
  expect(registrationPage.getAllByPlaceholderText('Full Name')).toMatchSnapshot();
  expect(registrationPage.getByText('Sign Up').tagName).toMatchSnapshot();
});
