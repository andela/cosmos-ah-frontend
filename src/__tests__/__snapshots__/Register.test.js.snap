// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Component 1`] = `
Array [
  <input
    placeholder="Full Name"
    type="text"
  />,
]
`;

exports[`renders the Component 2`] = `"BUTTON"`;
